src/file-manager/fm-desktop-icon-view.c

```diff
@@ -510,6 +553,13 @@ delayed_init (FMDesktopIconView *desktop_icon_view)
                              "done_loading",
                              G_CALLBACK (done_loading), desktop_icon_view, 0);
 
+    /*Handle the middle click on x11 */
```

Comment:
is it OK to move this here from `fm_desktop_icon_view_init()`? It probably is, but given the test below was totally wrong it hasn't been tested

Reply:
I have no idea and have tried to avoid deliberate changes in the x11 codepath. That would be for another PR is my guess, in case it ever had to be reverted. Any one commt in this would be combined with the rest in a squash and merge

Reply:
I will check to ensure that wasn't somehow moved by mistake and will move it back if it was

Reply:
Here's what I found: for some reason, running`  if (GDK_IS_X11_DISPLAY (gdk_display_get_default))`  in `fm_desktop_icon_view_init()` for some reason could not access the display and segfaulted. I returned that code to it's original location (so as to restore the old x11 order of things) and moved the x11 check to the middle click handling function itself and that works

Reply:
Moving the code back and putting the x11 check in fm_desktop_icon_view_handle_middle_click generated this warning when running in Wayland:

`sys:1: Warning: ../gobject/gsignal.c:2517: signal 'middle_click' is invalid for instance '0x5614b513e5c0' of type 'FMDesktopIconView'
`

Reply:
As described above, the problem that originally forced this to move was accidently using
` if (GDK_IS_X11_DISPLAY (gdk_display_get_default))`
instead of the intended
` if (GDK_IS_X11_DISPLAY (gdk_display_get_default()))`
and for some reason the compiler didn't stop the build and return an error on that